client/src/components/new-client-component/new-client-component.jsx

```diff
@@ -0,0 +1,72 @@
+import { useState } from "react";
+import {
+  Typography,
+  Stack,
+  Button,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+} from "@mui/material";
+import { Header, Form, Discard, AddButton } from "./new-client.styles";
+import ClientCard from "./ClientCard";
+
+function NewClient() {
+  const [clients, setClients] = useState([
```

Comment:
I like how you created this state in the parent component. However the data within it isn't being changed along with the fields. instead of mainting fullname, phonenumber and email as a state in each clientcard component, instead use the state you've created here. You could then pass the client data to each of the client cards as well as the setClients method (or some variation of these methods as props). 

The code for rendering a child component might end up looking something like this
`<ClientCard clientData={clients[i]} setClientData = {(newClientData) = setClientData(clients.slice(0, i).concat([newClientData]).concat(clients.slice(i + 1, -1))}`

Basically give each component the ability to change the data associated with it in the parent.